services/user/TokenSys/ui/src/queries.ts

```diff
@@ -61,3 +65,27 @@ export const useTransferHistory = (user?: string) => {
     const query = user && queryTransferHistory(user);
     return useGraphQLQuery("/graphql", query);
 };
+
+const querySharedBalances = () => `{
+  sharedBalances(last:1000) {
+    pageInfo {
+      hasNextPage
+      endCursor
+    }
+    edges {
+      node {
+        balance
+        key {
+          creditor
+          debitor
+          tokenId
+        }
+      }
+    }
+  }
+}`;
+
+export const useSharedBalances = (user?: string) => {
+    const query = querySharedBalances();
+    return useGraphQLQuery("/graphql", query);
+};
```

Comment:
The `user` parameter is unused. Remove it.

Out of curiosity, how does the query know _who_ to get shared balances for? I tested it, and it's not picking up other users' shared balances IIRC. So that's good.

Reply:
It does fetch all users shared balances (last 1000 records), making it return only user-related shared balances requires some backend work. I believe this task should address this if implemented https://github.com/orgs/gofractally/projects/4/views/1

Reply:
Ah, that makes sense. 👍